arangod/Aql/Query.cpp

```diff
@@ -1007,36 +1000,49 @@ QueryResult Query::explain() {
         _queryOptions.verbosePlans, _queryOptions.explainInternals,
         _queryOptions.explainRegisters == ExplainRegisterPlan::Yes);
 
-    if (_queryOptions.allPlans) {
-      result.data = std::make_shared<VPackBuilder>();
-      {
-        VPackArrayBuilder guard(result.data.get());
-
-        auto const& plans = opt.getPlans();
-        for (auto& it : plans) {
-          auto& pln = it.first;
-          TRI_ASSERT(pln != nullptr);
-
-          preparePlanForSerialization(pln);
-          pln->toVelocyPack(*result.data.get(), parser.ast(), flags);
+    std::unique_ptr<Optimizer> opt = nullptr;
```

Comment:
nit: doesn't need to be explicitly initialized to nullptr, as this happens automatically.
```suggestion
    std::unique_ptr<Optimizer> opt;
```